src/App.tsx

```diff
@@ -1,203 +1,113 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useEffect, useState } from 'react';
+import { deleteTodos, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+import { FilterOptions } from './enums/FilterOptions';
+import { ErrorMessages } from './enums/ErrorMessages';
+import { filteredTodos } from './utils/filteringTodos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterOptions>(FilterOptions.ALL);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessages.NO_ERRORS);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        const loadedTodos = await getTodos();
+
+        setTodos(loadedTodos);
+      } catch (error) {
+        setErrorMessage(ErrorMessages.LOADING_ERROR);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      const timer = setTimeout(() => {
+        setErrorMessage(ErrorMessages.NO_ERRORS);
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [errorMessage]);
+
+  const handleTodoDelete = (todoId: number) => {
+    deleteTodos(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.DELETING_ERROR);
+      });
+  };
+
+  const handleCompletedTodosDeleted = () => {
```

Comment:
1) `useCallback`? use it everywhere u need it
2) sounds like all the COMPLETED todos will be removed. Maybe - `handleDeleteSelectedTodos`?
```suggestion
  const handleDeleteSelectedTodos = () => {
```